### DISCUSSION TITLE
Change table row height programatically

### DISCUSSION THREAD

**Discussion by andreash** [palantir/blueprint#5742] [Q&A] [answered]
The row heights can be changed in the UI with the mouse. I would appreciate if there was an API to change the row heights programmatically as well. It would then be possible to use other form elements (with larger height than a normal text cell) inside the cells.

**Comment by adidahiya** [comment, marked as answer]
@andreash there are `<Table>` and `<Table2>` component instance methods to resize rows based on the content in each cell, try them out with something like this:

```ts
const tableRef = React.createRef<Table2>();

React.useLayoutEffect(() => {
  tableRef.current?.resizeRowsByTallestCell();
}, []);

return (
  <Table2 ref={tableRef} {...otherProps} />
);
```

Docs here: https://blueprintjs.com/docs/#table/api.instance-methods
